src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4722,10 +4490,8 @@ internal static List<CompletionResult> CompleteVariable(CompletionContext contex
                             }
                         }
 
-                        var completedName = (name.IndexOfAny(s_charactersRequiringQuotes) == -1)
-                                                ? prefix + provider + name
-                                                : prefix + "{" + provider + name + "}";
-                        AddUniqueVariable(hashedResults, results, completedName, name, tooltip);
+                        AddUniqueVariable(hashedResults, results,
+                            GenerateVariableCompletionText(provider + name, prefix, colon != -1), name, tooltip);
```

Comment:
I'm considering putting the call to GenerateVariableCompletionText inside AddUniqueVariable, because if the variable is not unique, there is no reason to determine if it needs braced.